m4/check_pkg_cuda.m4

```diff
@@ -39,19 +39,28 @@ AC_DEFUN([CHECK_PKG_CUDA], [
          cuda_ldpath="${cuda_realpath}/lib64"
          CUDA_LDFLAGS="-L${cuda_ldpath}"
          CUDA_CPPFLAGS="-isystem ${cuda_realpath}/include"
-         CUDA_LIBS="-l${cudart_lib} -lrt -ldl"
+         AS_IF([test "x${enable_cudart_dynamic}" = "xyes"],
+               [CUDA_LIBS="-lrt -ldl"],
+               [CUDA_LIBS="-l${cudart_lib} -lrt -ldl"])
          LDFLAGS="${CUDA_LDFLAGS} ${LDFLAGS}"
          LIBS="${CUDA_LIBS} ${LIBS}"
          CPPFLAGS="${CUDA_CPPFLAGS} ${CPPFLAGS}"
         ])
 
   AS_IF([test "${check_pkg_found}" = "yes"],
         [AC_SEARCH_LIBS(
-         [cudaGetDriverEntryPoint],
-         [${cudartlib}],
-         [],
-         [check_pkg_found=no],
-         [-ldl -lrt])])
+           [cudaGetDriverEntryPointByVersion],
```

Comment:
This code doesn't make much sense.  If we're dlopening libcudart, then there's no indication that the libcudart we're linking against will be the same we're running against.  I think you want to `AC_SEARCH_LIBS` if you're linking and `AC_CHECK_DECL` if you're going to use dlopen.

Also, the change from looking only for `cudaGetDriverEntryPoint` to either `cudaGetDriverEntryPoint` or `cudaGetDriverEntryPoint` should be its own commit.